ludwig/trainers/trainer_lightgbm.py

```diff
@@ -831,6 +831,9 @@ def is_coordinator(self) -> bool:
             return True
         return self.horovod.rank() == 0
 
+    def barrier(self):
```

Comment:
One small suggestion is that we add this to the `BaseTrainer` class, since hyperopt is (reasonably?) assuming that this method exists for all Ludwig trainers.

Reply:
I would probably go with the default implementation, as I expect most other trainers probably don't need to think about the horovod barrier.

If I'm implementing a base class with a methods that already have a default implementation, my assumption would be that the authors of the base class are rather confident that that default implementation is "reasonable" and unless I know what I'm doing, it's not necessary for me to think about the details of those methods.